models/Mapper/ResultMapper.php

```diff
@@ -0,0 +1,321 @@
+                    break;
+
+                case ResultResponseVariable::class:
+                    $variable = $this->createResponseVariable($itemVariable);
+                    break;
+
+                case ResultTemplateVariable::class:
+                default:
+                    throw new common_exception_NotImplemented('Qti Result parser cannot deals with "' . get_class($itemVariable) . '".');
+                    break;
+            }
+
+            $variable->setEpoch($datetime->getMicroseconds(true) . ' ' . $datetime->format('U'));
+            $variables[] = $variable;
+        }
+
+        return $variables;
+    }
+
+    /**
+     * Initialize a taoResultServer_models_classes_Variable based on ItemVariable
+     * - including identifier, cardinality, baseType
+     *
+     * @param ItemVariable $itemVariable
+     * @param taoResultServer_models_classes_Variable $variable
+     * @return taoResultServer_models_classes_Variable
+     * @throws common_exception_InvalidArgumentType
+     */
+    protected function createVariableFromItemVariable(ItemVariable $itemVariable, taoResultServer_models_classes_Variable $variable)
+    {
+        $variable->setIdentifier((string) $itemVariable->getIdentifier());
+        $variable->setCardinality(Cardinality::getNameByConstant($itemVariable->getCardinality()));
+
+        if (!is_null($itemVariable->getBaseType())) {
+            $variable->setBaseType(BaseType::getNameByConstant($itemVariable->getBaseType()));
+        }
+
+        return $variable;
+    }
+
+    /**
+     * Transfer all attributes of a ResultOutcomeVariable to taoResultServer_models_classes_OutcomeVariable
+     *
+     * @param ResultOutcomeVariable $itemVariable
+     * @return taoResultServer_models_classes_OutcomeVariable
+     * @throws common_exception_InvalidArgumentType
+     * @todo Implements Long Interpretation
+     * @todo Implements Mastery Value
+     * @todo Implements multiple Values
+     * @todo Implements View
+     * @todo Implements Interpretation
+     */
+    protected function createOutcomeVariable(ResultOutcomeVariable $itemVariable)
+    {
+        /** @var taoResultServer_models_classes_OutcomeVariable $variable */
+        $variable = $this->createVariableFromItemVariable(
+            $itemVariable,
+            new taoResultServer_models_classes_OutcomeVariable()
+        );
+
+        if ($itemVariable->hasValues()) {
+            $variable->setValue($this->serializeValueCollection($itemVariable->getValues()));
+        }
+
+        if ($itemVariable->hasNormalMaximum()) {
+            $variable->setNormalMaximum((string)  $itemVariable->getNormalMaximum()->getValue());
+        }
+
+        if ($itemVariable->hasNormalMinimum()) {
+            $variable->setNormalMinimum((string)  $itemVariable->getNormalMinimum()->getValue());
+        }
+
+        if ($itemVariable->hasView()) {
+            $this->logInfo('Qti Result Parser does not handle Outcome View');
```

Comment:
Are that out of scope of the task? Same question for all 4 sub-conditions above

Reply:
All todo are here to put under light feature supported by qti-sqk, but not handled by tao. It requires to be implemented but out of scope of this task